### PR TITLE
Refactorings

### PR DESCRIPTION
**Pull request by tklovett**
I like to explore a library by refactoring it as I step through -- here are the results of that process. Do you want any of these changes? Feel free to comment heavily with which changes you like or don't like, and I'll make those changes.

I'm fairly confident that I didn't change any truly public APIs, but it would be good to keep an eye on that as part of the review process.

### CODE REVIEW

**Comment by bbakerman** [comment]
So one of the big aspects of this refactor is changing the static builder method to new Builder

     FieldCollectorParameters collectorParameters = newParameters(executionContext.getGraphQLSchema(),operationRootType)
    FieldCollectorParameters collectorParameters = new FieldCollectorParameters.Builder()

The static method pattern is used everywhere and deliberately.

I am hesitant to accept this PR in this format since static method approach is where the code base is heading

There are some great little bits in this PR say around the `execute` setup but with all the new Builder code I dont think we should accept this.

File: src/main/java/graphql/GraphQL.java [hunk 1]
```diff
@@ -244,50 +244,108 @@ public ExecutionResult execute(String requestString, String operationName, Objec
      * @return result including errors
      */
     public ExecutionResult execute(ExecutionInput executionInput) {
```
Comment (bbakerman) [inline src/main/java/graphql/GraphQL.java]: I would rather see a new PR with the GraphQL class refactorings say then all the new Builder() changes

File: src/main/java/graphql/execution/Execution.java [hunk 2]
```diff
@@ -42,32 +40,24 @@ public Execution(ExecutionStrategy queryStrategy, ExecutionStrategy mutationStra
     }
 
     public ExecutionResult execute(ExecutionId executionId, GraphQLSchema graphQLSchema, Object context, Object root, Document document, String operationName, Map<String, Object> args) {
-        ExecutionContextBuilder executionContextBuilder = new ExecutionContextBuilder(new ValuesResolver(), instrumentation);
-        ExecutionContext executionContext = executionContextBuilder
+        final ExecutionContext executionContext = new ExecutionContextBuilder()
+                .valuesResolver(new ValuesResolver())
+                .instrumentation(instrumentation)
                 .executionId(executionId)
-                .build(graphQLSchema, queryStrategy, mutationStrategy, subscriptionStrategy, context, root, document, operationName, args);
```
Comment (bbakerman) [inline src/main/java/graphql/execution/Execution.java]: This has always bugged me that we have a builder but we dont use it completely

**Comment by andimarek** [comment]
@tklovett thanks for this PR and your effort to improve graphql-java.

But this PR is too unfocused and too broad to be merged. 

Please create separate PRs for each change you wanna do, so that we can discuss it there.

Thanks!

**Comment by tklovett** [comment]
@bbakerman No problem -- I was expecting y'all wouldn't want the `new Foo.Builder()` changes included. I can revert those and open a new PR.

For discussions sake, why is the `newFoo()` pattern preferred? In my personal opinion, it adds a level of indirection without encapsulating anything unnecessary details. When inlined, I know what type is being instantiated and what modifications are done to it. From reading the static method call, I don't know a) where the method is defined, b) what intermediate (Builder) type is being created, or c) what if any modifications are done to the Builder before being returned. Of course IDE tooling somewhat eases those things. Additionally, the extra method adds boilerplate that wouldn't be there otherwise.

Again, I'm just interested in the discussion here. I'll absolutely revert those changes before opening a new PR.

File: src/main/java/graphql/execution/FieldCollectorParameters.java [hunk 3]
```diff
@@ -37,12 +37,6 @@ private FieldCollectorParameters(GraphQLSchema graphQLSchema, Map<String, Object
         this.objectType = objectType;
     }
 
-    public static Builder newParameters(GraphQLSchema graphQLSchema, GraphQLObjectType objectType) {
```
Comment (tklovett) [inline src/main/java/graphql/execution/FieldCollectorParameters.java]: Would you like me to revert this change entirely and keep the method signature the same?

Or I can change it to

```
public static Builder newParameters() {
    return new Builder();
}
```

with the assertions moved to `public FieldCollectorParameters build() {...}`, as they are in this PR.

**Pull request closed without merging**